packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -19,18 +21,16 @@ import type { CanvasEvent, CommonInteractionProps, Selection } from './models';
 import { MouseButton } from './models';
 import { boundWorldPointToFOV, getModifierKeyArray } from './utils';
 
-interface SelectionState {
-  selection: Selection | undefined;
-  isCancelled?: boolean;
-}
-
 interface Props extends CommonInteractionProps {
   id?: string;
   transformSelection?: (selection: Selection) => Selection;
-  onSelectionStart?: (selection: Selection) => void;
-  onSelectionChange?: (selection: Selection | undefined) => void;
+  onSelectionStart?: () => void;
+  onSelectionChange?: (
+    selection: Selection | undefined,
+    rawSelection: Selection
```

Comment:
`onSelectionChange` (as well as the render prop below) now receives the effective selection object as first parameter (i.e. the transformed selection), and the raw selection object as second parameter. The effective selection is the most important information so it makes sense for it to come first.

Note that the default `transformSelection` callback is just a pass-through, so the effective selection is guaranteed to be defined when the raw selection is defined.